sharkfuser/benchmarks/run_benchmark.py

```diff
@@ -0,0 +1,317 @@
+    command: str,
+    driver_path: str,
+    output_dir: Path | None,
+    rocprof_args: list[str],
+    verbose: bool,
+    cmd_num: int,
+    use_tempdir: bool,
+) -> TimingStats:
+
+    driver_args = command.split()
+    if not driver_args:
+        if verbose:
+            print(f">>> Failed to parse command: {command}")
+        return TimingStats()
+
+    driver_cmd = [driver_path] + driver_args
+
+    # Use either temporary directory or persistent directory
+    if use_tempdir:
+        tmpdir_context = tempfile.TemporaryDirectory()
+        cmd_output_dir = Path(tmpdir_context.__enter__())
+    else:
+        tmpdir_context = None
+        cmd_output_dir = output_dir / f"command_{cmd_num}"
+        cmd_output_dir.mkdir(parents=True, exist_ok=True)
+
+    try:
+        rocprof_cmd = (
+            [
+                "rocprofv3",
+                "--output-format",
+                "csv",
+                "--output-directory",
+                str(cmd_output_dir),
+            ]
+            + rocprof_args
+            + ["--"]
+            + driver_cmd
+        )
+
+        if verbose:
+            print(f">>> {shlex.join(rocprof_cmd)}\n")
+
+        result = subprocess.run(
+            rocprof_cmd,
+            check=True,
+            capture_output=True,
+            text=True,
+        )
+
+        if verbose and result.stdout:
+            print(result.stdout)
+
+        stats = parse_rocprof_csv(cmd_output_dir)
+
+        if verbose:
+            print(
+                f">>> Stats: min={stats.min}, max={stats.max}, mean={stats.mean}, count={stats.count}"
+            )
+
+        return stats
+
+    except subprocess.CalledProcessError as e:
+        if verbose:
+            print(f">>> Command failed with exit code {e.returncode}")
+            if e.stderr:
+                print(f">>> stderr: {e.stderr}")
+        return TimingStats()
+    except Exception as e:
+        if verbose:
+            print(f">>> Exception: {e}")
+        return TimingStats()
+    finally:
+        # Cleanup temporary directory if used
+        if tmpdir_context is not None:
+            tmpdir_context.__exit__(None, None, None)
+
+
+def get_parser() -> argparse.ArgumentParser:
+    parser = argparse.ArgumentParser(
+        description="""
+Run Fusilli benchmarks with rocprofv3 profiling and aggregate results.
+
+Commands are read from a file (one per line). Each command is run through
+rocprofv3, and timing statistics are collected and written to a CSV file.
+
+Command format example:
+  conv --bf16 -n 16 -c 64 -H 48 -W 32 -k 64 -y 3 -x 3 -p 1 -q 1 -u 1 -v 1 -l 1 -j 1 --in_layout NHWC --out_layout NHWC --fil_layout NHWC --spatial_dim 2
+
+The script will:
+  1. Run each command under rocprofv3
+  2. Extract timing statistics from rocprof CSV outputs
+  3. Aggregate results into a single output CSV
+        """,
+        formatter_class=argparse.RawDescriptionHelpFormatter,
+    )
+
+    parser.add_argument(
+        "--commands-file",
```

Comment:
nit: consider adding a short option to this (like `-f`) for ease of not typing it all out